Source/WebCore/platform/mediastream/webrtcorg/PeerConnectionBackendWebRtcOrg.cpp

```diff
@@ -153,56 +286,94 @@ RefPtr<RTCSessionDescription> PeerConnectionBackendWebRtcOrg::pendingRemoteDescr
     return RefPtr<RTCSessionDescription>();
 }
 
-void PeerConnectionBackendWebRtcOrg::setConfiguration(RTCConfiguration& config, const MediaConstraints& constraints)
+void PeerConnectionBackendWebRtcOrg::setConfiguration(RTCConfiguration& rtcConfig, const MediaConstraints& constraints)
 {
-    WRTCInt::RTCConfiguration wrtcConfig;
-    for(auto& server : config.iceServers()) {
-        WRTCInt::RTCIceServer wrtcICEServer;
-        wrtcICEServer.credential = server->credential().utf8().data();
-        wrtcICEServer.username = server->username().utf8().data();
-        for(auto& url : server->urls()) {
-            wrtcICEServer.urls.push_back(url.utf8().data());
+    webrtc::PeerConnectionInterface::IceServers iceServers;
+    for (auto& server : rtcConfig.iceServers()) {
+        webrtc::PeerConnectionInterface::IceServer wrtcServer;
+        wrtcServer.username = server->credential().utf8().data();
+        wrtcServer.password = server->username().utf8().data();
+        for (auto& url : server->urls()) {
+            wrtcServer.urls.push_back(url.utf8().data());
         }
-        wrtcConfig.iceServers.push_back(wrtcICEServer);
+        iceServers.push_back(wrtcServer);
     }
-
-    WRTCInt::RTCMediaConstraints wrtcConstraints;
-    Vector<MediaConstraint> mediaConstraints;
-    constraints.getMandatoryConstraints(mediaConstraints);
-    for (auto& c : mediaConstraints) {
-        std::string name = c.m_name.utf8().data();
-        std::string value = c.m_value.utf8().data();
-        wrtcConstraints[name] = value;
-    }
-    mediaConstraints.clear();
-    constraints.getOptionalConstraints(mediaConstraints);
-    for (auto& c : mediaConstraints) {
-        std::string name = c.m_name.utf8().data();
-        std::string value = c.m_value.utf8().data();
-        wrtcConstraints[name] = value;
+    std::unique_ptr<WRTCMediaConstraints> cts(new WRTCMediaConstraints(constraints));
+    webrtc::PeerConnectionInterface::RTCConfiguration configuration;
+    configuration.servers = iceServers;
+    if (m_peerConnection == nullptr) {
+        m_peerConnection = m_peerConnectionFactory->CreatePeerConnection(
+               		configuration, cts.get(), nullptr, nullptr, this); 
+    } 
+    m_peerConnection->UpdateIce(iceServers, cts.get());
+    m_peerConnection->SetConfiguration(configuration);
```

Comment:
Looks like a merge/rebase artifact... Let's have peerconnection object creation/update to look like:

if (m_peerConnection == nullptr)  {
  m_peerConnection =  ... create ...
  m_peerConnection->SetConfiguration(configuration);
} else {
  m_peerConnection-> UpdateIce(iceServers, cts.get()));
}